src/IQFeed.CSharpApiClient/Lookup/Common/BaseLookupMessageHandler.cs

```diff
@@ -9,20 +10,32 @@ namespace IQFeed.CSharpApiClient.Lookup.Common
     public abstract class BaseLookupMessageHandler
     {
         private static readonly string EndMessagePattern = IQFeedDefault.ProtocolEndOfMessageCharacters + IQFeedDefault.ProtocolDelimiterCharacter;
+        private static readonly string NoDataIndicator = "E,!NO_DATA!";
+        private static readonly Regex ErrorPattern = new Regex(@"^E,.*,$");
+        private static readonly Regex ErrorPatternWithRequestId = new Regex(@"^.*,E,.*,$");        
 
-        protected static MessageContainer<T> ProcessMessages<T>(Func<string, T> parser, byte[] message, int count)
+        protected static MessageContainer<T> ProcessMessages<T>(Func<string, T> parser, byte[] message, int count, bool hasRequestId)
```

Comment:
I think I would prefer another method instead of adding new logic
ProcessMessagesWithRequestId --> this one focused to parse with request id.

Reply:
that will be more efficient, less if cases and also easier to test all different cases.

Reply:
there will be lot of duplicate code and potential for discrepancies later on

Reply:
let me think about this

Reply:
Please have a look at this version. I'd really hate to duplicate the entire message just to avoid a single if